src/server/pachyderm_test.go

```diff
@@ -1191,6 +1191,54 @@ func TestPipelineFailure(t *testing.T) {
 	require.True(t, strings.Contains(jobInfo.Reason, "datum"))
 }
 
+func TestEgressFailure(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode")
+	}
+	t.Parallel()
+	c, _ := minikubetestenv.AcquireCluster(t)
+
+	repo := tu.UniqueString(t.Name())
+	require.NoError(t, c.CreateRepo(repo))
+	pipeline := tu.UniqueString("egress")
+	_, err := c.PpsAPIClient.CreatePipeline(
+		c.Ctx(),
+		&pps.CreatePipelineRequest{
+			Pipeline: client.NewPipeline(pipeline),
+			Transform: &pps.Transform{
+				Image: tu.DefaultTransformImage,
+				Cmd:   []string{"bash"},
+				Stdin: []string{"cp -r /pfs/in/* /pfs/out/"},
+			},
+			Input: &pps.Input{Pfs: &pps.PFSInput{
+				Repo: repo,
+				Glob: "/",
+				Name: "in",
+			}},
+			Egress: &pps.Egress{
+				URL: "garbage-url",
+			},
+		},
+	)
+	require.NoError(t, err)
+	commit, err := c.StartCommit(repo, "master")
+	require.NoError(t, err)
+	require.NoError(t, c.PutFile(commit, "/test_table/0000", strings.NewReader("1,Foo\n2,Bar")))
+	require.NoError(t, c.FinishCommit(repo, "master", commit.ID))
+	var jobInfos []*pps.JobInfo
+	require.NoErrorWithinTRetry(t, time.Minute, func() error {
+		jobInfos, err = c.ListJob(pipeline, nil, -1, true)
+		require.NoError(t, err)
+		if len(jobInfos) != 1 {
+			return errors.Errorf("expected 1 jobs, got %d", len(jobInfos))
```

Comment:
pretty sure this will never be true with the current ordering, you'll get a job when you create the pipeline, then a second one when you create the commit. You can either create the commit with data first or expect two jobs

I'm going to make a PR to fix `NoErrorWithinTRetry` to actually show the error you care about

Reply:
Thanks, I moved the commit before the pipeline